File: src/BreakfastProvider.Api/Models/Responses/IngredientUsageResponse.cs
namespace BreakfastProvider.Api.Models.Responses;

public class IngredientUsageResponse
{
    public string UsageId { get; set; } = string.Empty;
    public string IngredientName { get; set; } = string.Empty;
    public decimal QuantityUsed { get; set; }
    public string Unit { get; set; } = string.Empty;
    public string RecipeName { get; set; } = string.Empty;
    public DateTime RecordedAt { get; set; }
}

public class IngredientUsageSummaryResponse
{
    public string IngredientName { get; set; } = string.Empty;
    public decimal TotalQuantityUsed { get; set; }
    public string Unit { get; set; } = string.Empty;
    public int RecordCount { get; set; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/response/IngredientUsageResponse.java
package io.lemonlion.breakfast.model.response;

import java.math.BigDecimal;
import java.time.Instant;

/** Twin of C# {@code IngredientUsageResponse}. */
public record IngredientUsageResponse(
        String usageId, String ingredientName, BigDecimal quantityUsed, String unit, String recipeName,
        Instant recordedAt) {
}
